### PR TITLE
Fix test case "client-operation-group" for python

### PR DESCRIPTION
**Pull request by msyyc**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp [hunk 1]
```diff
@@ -54,14 +54,14 @@ namespace Client.Structure.ClientOperationGroup {
...
-  "csharp,java,python"
+  "csharp,java"
 )
 @client(
   {
     name: "SecondClient",
     service: Client.Structure.Service,
   },
-  "javascript,go"
+  "javascript,go,python"
```
Comment (msyyc) [inline packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp:64]: Python SDK doesn't permit client name has namespace prefix.

Comment (msyyc) [reply packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp:64]: @tadelesh @iscai-msft please help review

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp (hunk 1 above)
Comment (tadelesh) [reply packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp:64]: > Python SDK package name, namespace and package folder are one-one mapping. If add sub-namespace, package folder shall also have related change which makes the folder complicated to understand for users.

do you have a link to the generated code?

Comment (msyyc) [reply packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp:64]: Without this PR, python emitter will crash for this case.

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp:64]: approve this change, but i believe we need to discuss this scenario in python.

**Review by tadelesh — APPROVED** [review]

File: packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp (hunk 1 above)
Comment (iscai-msft) [reply packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp:64]: this is related to the overarching namespace changes we're looking into, see @srnagar 's proposal [here](https://gist.github.com/srnagar/a695faa1a4d8a5dbf44af826f05916f0). Python should start following this structure, which is something that all languages will start working on

Comment (msyyc) [reply packages/cadl-ranch-specs/http/client/structure/client-operation-group/client.tsp:64]: I see and create https://github.com/Azure/autorest.python/issues/2795 to track

**Pull request closed without merging**